controllers/deadmanssnitchintegration/event_handlers_test.go

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deadmanssnitchintegration
+
+import (
+	"testing"
+
+	deadmanssnitchv1alpha1 "github.com/openshift/deadmanssnitch-operator/api/v1alpha1"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func Test_enqueueRequestForClusterDeployment_toRequests(t *testing.T) {
+	scheme := runtime.NewScheme()
+	s := runtime.SchemeBuilder{
+		corev1.AddToScheme,
+		hivev1.AddToScheme,
+		deadmanssnitchv1alpha1.AddToScheme,
+	}
+	assert.Nil(t, s.AddToScheme(scheme))
+
+	tests := []struct {
+		name             string
+		obj              client.Object
+		pdiObjs          []runtime.Object
+		expectedRequests int
+	}{
+		{
+			name:             "empty ClusterDeployment",
+			obj:              &hivev1.ClusterDeployment{},
+			pdiObjs:          []runtime.Object{},
+			expectedRequests: 0,
+		},
+		{
+			name: "empty ClusterDeployment",
+			obj: &hivev1.ClusterDeployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"pdiWatching": "clusterDeployment1",
```

Comment:
Though it ultimately probably doesn't matter from a testing perspective, renaming this to something that doesn't carry pagerduty-operator connotations might be better. 

```suggestion
						"dmsiWatching": "clusterDeployment1",
```

Same goes for all other `pdi` references.

Reply:
Thanks for catching that. I will get those fixed.